apps/worker/services/lock_manager.py

```diff
@@ -83,8 +100,24 @@ def locked(self, lock_type: LockType, retry_num=0):
                     },
                 )
         except LockError:
-            max_retry = 200 * 3**retry_num
-            countdown = min(random.randint(max_retry // 2, max_retry), 60 * 60 * 5)
+            max_retry_cap = 60 * 60 * 5  # 5 hours
+            max_retry_unbounded = 200 * 3**retry_num
+            countdown_unbounded = random.randint(
+                max_retry_unbounded // 2, max_retry_unbounded
+            )
+            countdown = min(countdown_unbounded, max_retry_cap)
+
+            if max_retries is not None and retry_num >= max_retries:
+                log.error(
```

Comment:
@thomasrockhu-codecov if we hit max retry,  it should hit here. I can't find this log though. So, either we aren't,  or something is wrong.